classes/rule.php

```diff
@@ -827,8 +827,31 @@ private function check_request_length($mod) {
      * @return boolean
      */
     private function check_elapsed_length($request, $currenttime) {
-        $delta = utility::calculate_weekdays_elapsed($request->request->timestamp, $currenttime);
 
+        // If this request has been reopened, check against the reopened date, not the original request date.
+        global $DB;
+
+        $sql = "SELECT hs.timestamp
+                FROM {local_extension_hist_state} hs
+                WHERE state = :reopened AND requestid = :requestid
+                ORDER BY hs.timestamp ";
+
+        $params = [
+            'reopened' => state::STATE_REOPENED,
+            'requestid' => $request->requestid,
+        ];
+
+        $lastreopeneddate = $DB->get_fieldset_sql($sql, $params);
+
+        if ($lastreopeneddate) {
+            $checkdate = (int)max($lastreopeneddate);
```

Comment:
We could simplify things a _little_ bit here by using get_field_select with MAX(timestamp) for our query. Then it wouldn't be necessary to use max in PHP code (at the moment we're doing the sorting logic once in the query and then again in PHP).